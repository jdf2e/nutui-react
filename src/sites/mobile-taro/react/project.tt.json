{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
