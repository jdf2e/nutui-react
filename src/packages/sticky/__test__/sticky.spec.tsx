import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Sticky from '@/packages/sticky'
import Button from '@/packages/button'
import Cell from '@/packages/cell'

test('should match snapshot', () => {
  const { asFragment } = render(
    <Cell>
      <Sticky>
        <Button type="primary">吸顶</Button>
      </Sticky>
    </Cell>
  )
  expect(asFragment()).toMatchSnapshot()
})

test('should sticky to top after scroll', () => {
  const { container } = render(
    <>
      <Cell>
        <Sticky>
          <Button type="primary">吸顶</Button>
        </Sticky>
      </Cell>
    </>
  )
  expect(container.querySelector('.nut-sticky--box')).toHaveAttribute(
    'style',
    'top: 0px; z-index: 900;'
  )
})
test('should sticky to bottom after scroll', () => {
  const { container } = render(
    <>
      <Cell>
        <Sticky position="bottom" threshold={20}>
          <Button type="primary">吸底</Button>
        </Sticky>
      </Cell>
    </>
  )
  expect(container.querySelector('.nut-sticky--box')).toHaveAttribute(
    'style',
    'bottom: 20px; z-index: 900; position: fixed;'
  )
})
