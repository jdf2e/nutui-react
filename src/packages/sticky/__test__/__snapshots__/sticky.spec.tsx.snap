// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nut-cell"
    style="border-radius: 6px; align-items: flex-start;"
  >
    <div
      class="nut-sticky"
    >
      <div
        class="nut-sticky-box"
        style="top: 0px; z-index: 900;"
      >
        <button
          class="nut-button nut-button-primary nut-button-primary-solid nut-button-normal nut-button-round"
          type="button"
        >
          <div
            class="nut-button-wrap"
          >
            <div
              class=""
            >
              吸顶
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
